### PR TITLE
python3Packages.pre-commit-po-hooks: init at 1.7.3

### PR DESCRIPTION
**Pull request by yajo**
## Description of changes
Packaged it.

@moduon MT-1075

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/pre-commit-po-hooks/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+, pytestCheckHook
+, pythonOlder
+, setuptools
+}:
+buildPythonPackage rec {
+  pname = "pre-commit-po-hooks";
+  version = "1.7.3";
+  pyproject = true;
+
+  disabled = pythonOlder "3.6";
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pre-commit-po-hooks/default.nix:13]:
```suggestion
```
We don't have that old Python versions in nixpkgs.

File: pkgs/development/python-modules/pre-commit-po-hooks/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{ lib
+, buildPythonPackage
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pre-commit-po-hooks/default.nix:2]: Please format with the official formatter.

File: pkgs/development/python-modules/pre-commit-po-hooks/default.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+  pythonImportsCheck = [
+    "pre_commit_po_hooks"
+  ];
+
+  meta = with lib; {
+    description = "Hooks for pre-commit useful working with PO files";
+    homepage = "https://github.com/mondeja/pre-commit-po-hooks";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ yajo ];
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pre-commit-po-hooks/default.nix:38]:
```suggestion
  meta = {
    description = "Hooks for pre-commit useful working with PO files";
    homepage = "https://github.com/mondeja/pre-commit-po-hooks";
    license = lib.licenses.bsd3;
    maintainers = with lib.maintainers; [ yajo ];
```

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

**Comment by yajo** [comment]
All attended.

File: pkgs/development/python-modules/pre-commit-po-hooks/default.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  setuptools,
+}:
+buildPythonPackage rec {
+  pname = "pre-commit-po-hooks";
+  version = "1.7.3";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "mondeja";
+    repo = pname;
```
Comment (FliegendeWurst) [inline pkgs/development/python-modules/pre-commit-po-hooks/default.nix:15]:
```suggestion
    repo = "pre-commit-po-hooks";
```
#277994

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/development/python-modules/pre-commit-po-hooks/default.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "mondeja";
+    repo = "pre-commit-po-hooks";
+    rev = "v${version}";
+    hash = "sha256-wTmcV8KkoQLuK4EWDt0pbp+EQJRatxnQYeBfikK+vlA=";
+  };
+
+  nativeBuildInputs = [ pytestCheckHook ];
```
Comment (natsukium) [inline pkgs/development/python-modules/pre-commit-po-hooks/default.nix:20]:
```suggestion
  nativeCheckInputs = [ pytestCheckHook ];
```

**Review by natsukium — CHANGES_REQUESTED** [review]

**Comment by natsukium** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 295568`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python311Packages.pre-commit-po-hooks</li>
    <li>python312Packages.pre-commit-po-hooks</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python311Packages.pre-commit-po-hooks</li>
    <li>python312Packages.pre-commit-po-hooks</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python311Packages.pre-commit-po-hooks</li>
    <li>python312Packages.pre-commit-po-hooks</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>python311Packages.pre-commit-po-hooks</li>
    <li>python312Packages.pre-commit-po-hooks</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**